app/include/zmk/event_manager.h

```diff
@@ -14,16 +14,16 @@ struct zmk_event_type {
     const char *name;
 };
 
-struct zmk_event_header {
+typedef struct {
```

Comment:
This seemed to really complicate things, since it requires the generated structs now be moved to using a pointer to a `zmk_event_t`, which then requires two phase alloc/free, which changes the API everywhere, etc.

Any strong reason we need to *enforce* opaque for this at this point? Not sure the value *yet*, versus incrementally improving this.

Reply:
Probably not necessary unless we start getting reviews where people are accidentally misusing the event types.